examples/postgresql-public-iam/main.tf

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  version = "~> 3.22"
+}
+
+provider "google-beta" {
+  version = "~> 3.5"
+}
+
+provider "null" {
+  version = "~> 2.1"
+}
+
+provider "random" {
+  version = "~> 2.2"
+}
+
+module "postgresql-db" {
+  source               = "../../modules/postgresql"
+  name                 = var.db_name
+  random_instance_name = true
+  database_version     = "POSTGRES_9_6"
+  project_id           = var.project_id
+  zone                 = "us-central1-c"
+  region               = "us-central1"
+  tier                 = "db-f1-micro"
+
+  deletion_protection = false
+
+  ip_configuration = {
+    ipv4_enabled        = true
+    private_network     = null
+    require_ssl         = true
+    authorized_networks = var.authorized_networks
+  }
+
+  database_flags = [
+    {
+      name  = "cloudsql.iam_authentication"
+      value = "On"
+    },
+  ]
+
+  additional_users = [
+    {
+      name     = "tftest2"
+      password = "abcdefg"
+      host     = "localhost"
+    },
+    {
+      name     = "tftest3"
+      password = "abcdefg"
+      host     = "localhost"
+    },
+  ]
+
+  # Supports creation of both IAM Users and IAM Service Accounts
+  iam_users = [
+    {
+      email         = var.cloudsql_pg_sa
+      is_account_sa = true
```

Comment:
Would it be possible to do this automatically? SA would be of the form `...gserviceaccount.com` so perhaps `is_sa = trimsuffix(var. email"gserviceaccount.com")==var. email`

Reply:
That's actually being done in the model [here](https://github.com/terraform-google-modules/terraform-google-sql-db/pull/218/files#diff-5af99345140d405f1b9683608683cd8e45b6cce368ba0446706651ca3a92f041R181). I wanted to keep the input as simple as possible and let the module do the trimming/formatting etc. Let me know what you think.

Reply:
Rethinking this and I think you have a valid point. If the email is an SA or not can be calculated through logic and I'd rather put that logic in the module as well and let the user just provide the email. That way we can get rid of that that attribute as well. I'll make that change.

Reply:
@g-awmalik I was thinking if we can detect if its an SA or not without needing user input.

My idea was if `trimsuffix(each.value.email, ".gserviceaccount.com") != each.value.email` that would mean that `trimsuffix` was able to trim `.gserviceaccount.com` which means it is a service account. If both are equal that means `trimsuffix` failed to trim because no  ".gserviceaccount.com" suffix and hence user email. WDYT?

```tf
name =  trimsuffix(each.value.email, "gserviceaccount.com") != each.value.email ? ( trimsuffix(each.value.email, ".gserviceaccount.com")    ) : (    each.value.email  )
```

Reply:
Yeah totally agree. I think we crossed each other in our latest responses. I get it and I'll make that change. The user will only provide an email with that.

Reply:
done.